### PR TITLE
Do not call search::indexation at each entity creation

### PR DESCRIPTION
**Pull request by jocel1**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | see title
| Type?         | improvement
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4896
| How to test?  | should speed up a bit the current install. Check if the search index is properly computed at the end of the install + demo products

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8789)

### CODE REVIEW

File: classes/ConfigurationTest.php [hunk 1]
```diff
@@ -264,83 +264,31 @@ public static function test_dir($relative_dir, $recursive = false, &$full_report
...
+        static $recursiveTests = [
+            'test_img_dir',
+            'test_module_dir',
+            'test_mails_dir',
+            'test_translations_dir',
+            'test_config_sf2_dir'
+        ];
 
-        return file_exists($file) && is_writable($file);
+        return ConfigurationTest::test_dir($arguments[0], in_array($name, $recursiveTests));
```
Comment (Quetzacoalt91) [inline classes/ConfigurationTest.php]: I'm not in favor of this kind of magic call (`__callStatic`), this make the project more difficult to understand. :/

Comment (jocel1) [reply classes/ConfigurationTest.php]: you're right, I've just reverted those changes. I would be great to cleanup a bit this file later though (some test_ functions are not used anymore)

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by marionf** [comment]
Hello @jocel1 

The search no longer displays the products, even after re-index them.
pr8789

But if I click on enter, the product is well displayed.

**Comment by jocel1** [comment]
Hi @marionf!

I don't think this bug is related to my changes, it should not impact the front display, and I didn't change the reindexing function, so it should not impact a direct reindexing. Could you double check on your side if you have the problem without my changes?

Thanks!

**Comment by marionf** [comment]
@jocel1 
I don't have this issue on develop branch
capture du 2018-02-26 15-20-12

**Comment by jocel1** [comment]
Hi @marionf ,

I also have the issue on the develop branch , it's related to the following code in themes/classic/templates/catalog/_partials/miniatures/product.tpl:

```
        {block name='product_name'}
          {if $page.page_name == 'index'}
            <h3 class="h3 product-title" itemprop="name"><a href="{$product.url}">{$product.name|truncate:30:'...'}</a></h3>
          {else}
            <h2 class="h3 product-title" itemprop="name"><a href="{$product.url}">{$product.name|truncate:30:'...'}</a></h2>
          {/if}
        {/block}
```

It complains about the $page index which is undefined.

**Comment by mickaelandrieu** [comment]
Thank you @jocel1!

**Pull request merged**